client/components/data/domain-management/email/index.jsx

```diff
@@ -48,84 +51,125 @@ function getStateFromStores( props ) {
 	};
 }
 
-const EmailData = React.createClass( {
-	displayName: 'EmailData',
+export class EmailData extends Component {
+	static propTypes = {
+		context: PropTypes.object.isRequired,
+		fetchSitePlans: PropTypes.func.isRequired,
+		products: PropTypes.object.isRequired,
+		selectedDomainName: PropTypes.string,
+		selectedSite: PropTypes.object.isRequired,
+		sitePlans: PropTypes.object.isRequired,
+		googleAppsUsers: PropTypes.array.isRequired,
+		googleAppsUsersLoaded: PropTypes.bool.isRequired,
+	};
 
-	propTypes: {
-		component: React.PropTypes.func.isRequired,
-		context: React.PropTypes.object.isRequired,
-		productsList: React.PropTypes.object.isRequired,
-		selectedDomainName: React.PropTypes.string,
-		sitePlans: React.PropTypes.object.isRequired,
-		sites: React.PropTypes.object.isRequired,
-		googleAppsUsers: React.PropTypes.array.isRequired,
-		googleAppsUsersLoaded: React.PropTypes.bool.isRequired
-	},
+	constructor( props ) {
+		super( props );
 
-	mixins: [ observe( 'productsList' ) ],
+		props.loadDomainsAndSitePlans();
```

Comment:
This is kind of ugly because the original component was doing it @gziolo. I like what you suggested @gwwar and will try and pull in the site plans query component.

It's hard for me making the subjective call between _cleanup component_ and _eliminate sites-list_.